r/R/dplyr.R

```diff
@@ -434,6 +438,77 @@ build_function_list <- function(FUN) {
   )
 }
 
+arrow_r_string_replace_function <- function(FUN, max_replacements) {
+  function(pattern, replacement, x, ignore.case = FALSE, fixed = FALSE) {
+    if (ignore.case) {
+      if (fixed) {
+        pattern <- gsub("\\E", "\\e", pattern, fixed = TRUE)
```

Comment:
Could you leave some comments and/or links to the relevant regexp/re2 docs that explain why you're making these modifications?

Reply:
Added in db828beec72f5cac9e673bf4e3305c077e1bad5d